docs(llms): UX guidance for C–H components (#4542)
coveo/react-vapor: packages/llms/src/components/DatePickerInput.md

- `description` and `error` props SHOULD be used to communicate supplementary information and validation feedback respectively — these are rendered in accessible positions relative to the input.
- The calendar popover is keyboard-navigable: arrow keys move between days, Enter confirms selection, Escape closes the popover.

## Content guidance

- Use a `placeholder` that indicates the expected format (e.g. `"Pick a date"`) so users understand what the field expects before interacting.
- Labels SHOULD use noun phrases rather than instructions (e.g. "Start date" not "Select a start date").
- Error messages SHOULD be specific: "Start date must be before end date" rather than "Invalid date".

## Common anti-patterns

- Setting `numberOfColumns` without `type="range"` — the multi-column layout is only useful when selecting a range.
- Omitting `clearable` on an optional date field, forcing users to reload or use workarounds to remove a value.

## What an AI agent should understand

- `DatePickerInput` is a thin Plasma wrapper around `@mantine/dates` `DatePickerInput`. All Mantine props are available.
- `numberOfColumns` and `columnsToScroll` are only meaningful with `type="range"`.

> - `presets` works with any `type`: each preset is `{label: string; value}`, where `value` is a single ISO date string (`YYYY-MM-DD`) for `type="default"` and a `[start, end]` tuple for `type="range"`.

Reviewer: ```suggestion
- `presets` works with any `type`: for `type="default"` each preset is `{label: string; value: string}`, for `type="range"` each preset is `{label: string; value: [string, string]`. The `value` is an ISO date string (`YYYY-MM-DD`).
```

Author: The value-shape detail now lives on the API-reference `presets` prop (default = single ISO string, range = `[start, end]` tuple) ✅

> - Plasma registers a `displayName` for tooling but adds no additional logic beyond the Mantine base component.

Reviewer: That's internal, I would omit mentioning it
```suggestion
```

Author: Resolved — that line was in the "What an AI agent should understand" section, which we removed across the docs.

# API reference

## Props

> Extends: `DatePickerInputProps` from `@mantine/dates`. No additional Plasma-specific props beyond the Mantine base component.

Key props relevant to Plasma usage patterns:

**`type`** `'default' | 'multiple' | 'range'` · optional · default: `'default'` — Controls selection mode.

> **`presets`** `Array<{label: string; value: [string, string]}>` · optional — Shortcut range buttons displayed beside the calendar. Only meaningful with `type="range"`.

Reviewer: ```suggestion
**`presets`** `Array<{label: string; value: string}> | Array<{label: string; value: [string, string]}>` · optional — Shortcut range buttons displayed beside the calendar. The structure depends on the `type`. 
```

Author: Applied ✅
